ocs_ci/utility/kms.py

```diff
@@ -123,13 +129,22 @@ def __init__(self):
         )
         self.kmsid = None
         # Base64 encoded (with padding) token
-        self.vault_path_token = None
+        self.vault_path_token = Vault.vault_path_token
         self.vault_policy_name = None
         self.vault_kube_auth_path = "kubernetes"
         self.vault_kube_auth_role = constants.VAULT_KUBERNETES_AUTH_ROLE
         self.vault_kube_auth_namespace = None
         self.vault_cwd_kms_sa_name = constants.VAULT_CWD_KMS_SA_NAME
 
+    @staticmethod
+    def set_vault_token(self):
+        """
+        This is a static method that will set the token value.
+        """
+        if self.vault_path_token is None:
+            self.vault_path_token = self.generate_vault_token()
+        return self.vault_path_token
+
```

Comment:
A staticmethod won't have a `self` to reference, we want to use a `classmethod` here. 

```suggestion
    @classmethod
    def set_vault_token(cls):
        """
        This is a class method that will set the token value.
        """
        if cls.vault_path_token is None:
            cls.vault_path_token = cls.generate_vault_token()
        return cls.vault_path_token

```

Reply:
Hello, we tried the class method route which ended up causing failure in the verification job thus has to reverse back to the static method here.

Reply:
We are calling generate token function inside set token token function that is where the problem happens. This static method is being a simple solution here.

Reply:
Why does it need to be a staticmethod? You are circumventing it by passing a `self` reference from another method anyways.

Reply:
If I'm understanding the issue, it seems like calling `generate_vault_token()` from inside this as a classmethod is the issue. I'm assuming here, but that is most likely because we have it defined as a standard method (with a reference to self) and we might be calling this code before there is an object instantiated. Looking at `generate_vault_token`, it doesn't actually use the `self` reference, and as such could be a `@staticmethod`, which should fix that issue. 

Since we're using a class level variable here, and calling what is effectively a static method, I think this is a perfect use case to make `set_vault_token` a classmethod.

Please let me know if there was another issue I'm not seeing, but this solve the issue of calling `generate_vault_token` from within the `set_vault_token` classmethod.

Reply:
I took another look at the code and I think I'm understanding the issue. We are calling `generate_vault_token` which is a standard method with a reference to an instance of the class. If we are calling this class level method without having instantiated an actual object, then calling this method will fail. In this case, a class method might make things more difficult.

That being said, this should not be a `@staticmethod`. A static method is something that could be defined outside of the class and work the same, without any references to the class or an instance of the class (self). If we are going to circumvent this and manually pass a reference to the instance of the class then this should simply be a standard method with no `@staticmethod` decorator.

I will update my requested changes.